subsys/bluetooth/host/gatt.c

```diff
@@ -2744,14 +2745,17 @@ static u16_t parse_service(struct bt_conn *conn, const void *pdu,
 		       start_handle, end_handle, bt_uuid_str(&u.uuid));
 
 		if (params->type == BT_GATT_DISCOVER_PRIMARY) {
-			attr.uuid = BT_UUID_GATT_PRIMARY;
+			memcpy(&uuid_svc, BT_UUID_GATT_PRIMARY,
+			       sizeof(uuid_svc));
 		} else {
-			attr.uuid = BT_UUID_GATT_SECONDARY;
+			memcpy(&uuid_svc, BT_UUID_GATT_SECONDARY,
+			       sizeof(uuid_svc));
 		}
 
 		value.end_handle = end_handle;
 		value.uuid = &u.uuid;
 
+		attr.uuid = (struct bt_uuid *)&uuid_svc;
```

Comment:
`attr.uuid = &uuid_srv.uuid;` might be a bit cleaner since it avoids the typecast